routes/event.js

```diff
@@ -55,6 +56,48 @@ router.get('/guests/:event_id',isLoggedIn, isAdminUser, function(req, res) {
 	});
 });
 
+router.get('/guests/send/:event_id', isLoggedIn, isAdminUser, function(req, res) {
+	eventMapper.findEventBy_event_id(req.params.event_id, function(err, result) {
+		if (err) {
+			res.send(err);
+		}
+
+		res.render('sendGuestDetails', {result, err: req.flash('err'),succ: req.flash('succ')});
+	});
+});
+
+router.post('/guests/send/:event_id', isLoggedIn, isAdminUser, function(req, res) {
+	eventMapper.findEventBy_event_id(req.params.event_id, function(err, result) {
+		if (!result) {
+			req.flash('err', 'Event not found');
+		}
+		else if (err) {
+			res.send(err);
+		}
+		else
+		{
+			var details = 'The following is a specification of guest dietary restrictions, access requirements and table arrangements for the upcoming event, ' + result.title + ' on ' + result.date + ':\n\n';
+
+			addGuestDetails(result, details, function(guestDetails) {
+				addTableDetails(result, guestDetails, function(guestAndTableDetails) {	
```

Comment:
Would it possible to only concat guest details to details after the callback rather than in the function? This means if guest details are null (i.e no need to send email since itll be empty) we can not go any further and return flash message.

Reply:
Since your code essentially always updates with what it had previously at every iteration it means the callback(guestDetails) will always have at least the details string

Reply:
**If you still think we should send the email (even though no requirements for any guests) then maybe just a condtion which adds another hardcoded line like when the table info is not present